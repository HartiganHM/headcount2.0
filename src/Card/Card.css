.card {
  align-items: center;
  border: 1px solid gray;
  display: flex;
  flex: 1;
  flex-direction: column;
  height: 16rem;
  padding: 1rem;
  margin: 1rem;
  text-align: center;
  max-width: 12rem;
  min-width: 12rem; }
  .card .district-name {
    font-size: 1.2rem; }
  .card .district-data {
    font-size: 0.8rem;
    text-align: left; }
    .card .district-data .greaterThan {
      color: blue; }
    .card .district-data .lessThan {
      color: red; }

.selected {
  background-color: rgba(0, 197, 0, 0.37); }
