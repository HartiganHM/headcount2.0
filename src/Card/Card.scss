.card {
    align-items: center;
    border: 1px solid gray;
    display: flex;
    flex: 1;
    flex-direction: column;
    height: 16rem;
    padding: 1rem;
    margin: 1rem;
    text-align: center;
    max-width: 12rem;
    min-width: 12rem;

    .district-name {
        font-size: 1.2rem;
    }

    .district-data {
        font-size: 0.8rem;
        text-align: left;

        .greaterThan {
            color: blue;
        }

        .lessThan {
            color: red;
        }
    }
}
