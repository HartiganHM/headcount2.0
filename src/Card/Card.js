import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';

const Card = ({ district, data }) => {
  const districtData = Object.keys(data).map(year => {
    return (
      <li key={year}>
        {year}: {data[year]}
      </li>
    );
  });

  return (
    <div className='card'>
      <span className='district-name'>{district}</span>
      <ul className='district-data'>{districtData}</ul>
    </div>
  );
};

Card.propTypes = {
  district: PropTypes.string,
  data: PropTypes.object
};

export default Card;
