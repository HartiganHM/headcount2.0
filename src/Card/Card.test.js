import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card Tests', () => {
  let renderedCard;
  let mockDistrict;
  let mockData;
  let mockFunc;
  let mockType;

  beforeEach(() => {
    mockDistrict = 'colorado';
    mockData = { 2007: 0.88 };
    mockFunc = jest.fn();
    mockType = 'card';
    let { district, data } = mockData;
    renderedCard = shallow(
      <Card
        district={mockDistrict}
        data={mockData}
        clickFunction={mockFunc}
        type={mockType}
      />
    );
  });

  it('Card should exist', () => {
    expect(renderedCard).toBeDefined();
  });

  it('Card should match the snapshot', () => {
    expect(renderedCard).toMatchSnapshot();
  });

  it('Card should have a different type with new data', () => {
    expect(renderedCard.find('.high-score').length).toEqual(1);

    mockData = { 2005: 0.1 };
    renderedCard = shallow(<Card data={mockData} />);
    expect(renderedCard.find('.low-score').length).toEqual(1);
  });

  it('Card should have a list that matches number of years/ranges', () => {
    expect(renderedCard.find('.high-score').length).toEqual(1);

    mockData = { 2005: 0.8, 2006: 0.85, 2007: 0.9 };
    renderedCard = shallow(<Card data={mockData} />);

    expect(renderedCard.find('.high-score').length).toEqual(3);

    mockData = { 2005: 0.1, 2006: 0.3, 2007: 0.6, 2008: 0.9 };
    renderedCard = shallow(<Card data={mockData} />);

    expect(renderedCard.find('.high-score').length).toEqual(1);
    expect(renderedCard.find('.mid-high-score').length).toEqual(1);
    expect(renderedCard.find('.mid-low-score').length).toEqual(1);
    expect(renderedCard.find('.low-score').length).toEqual(1);
  });

  it('Card should send district name on click', () => {
    renderedCard.simulate('click');
    expect(mockFunc.mock.calls).toEqual([[mockDistrict]]);
  });
});
