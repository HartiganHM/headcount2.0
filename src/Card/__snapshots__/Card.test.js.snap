exports[`Card Tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "2007": 0.88,
          }
    }
    district="place" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="district-name">
          place
</span>,
        <ul
          className="district-data">
          <li
                    className="greaterThan">
                    2007
                    : 
                    0.88
          </li>
</ul>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "place",
          "className": "district-name",
        },
        "ref": null,
        "rendered": "place",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="greaterThan">
              2007
              : 
              0.88
</li>,
          ],
          "className": "district-data",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "2007",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2007",
                ": ",
                0.88,
              ],
              "className": "greaterThan",
            },
            "ref": null,
            "rendered": Array [
              "2007",
              ": ",
              0.88,
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="district-name">
            place
</span>,
          <ul
            className="district-data">
            <li
                        className="greaterThan">
                        2007
                        : 
                        0.88
            </li>
</ul>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "place",
            "className": "district-name",
          },
          "ref": null,
          "rendered": "place",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="greaterThan">
                2007
                : 
                0.88
</li>,
            ],
            "className": "district-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "2007",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2007",
                  ": ",
                  0.88,
                ],
                "className": "greaterThan",
              },
              "ref": null,
              "rendered": Array [
                "2007",
                ": ",
                0.88,
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
